libcloud/common/base.py

```diff
@@ -926,8 +926,13 @@ def __init__(self, key, secret=None, secure=True, host=None, port=None,
         self.secure = secure
         args = [self.key]
 
-        if self.secret is not None:
-            args.append(self.secret)
+        try:
+            if secret is not None or \
+               issubclass(self.connectionCls, ConnectionUserAndKey):
+                args.append(self.secret)
+        except TypeError:
```

Comment:
To take of test cases, where the Connection Class used is of type MockHttp. The check issubclass fails for the same.